### ISSUE TITLE
Validate email addresses upon registration

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#621]
# TL;DR
Validate email addresses upon registration to ensure we can actually reach participants and that researchers are associated with their stated email address.

# Narrative
As a new user, I may make a typo in my email address but not realize that email with compensation will not reach me. 

As a researcher, I want assurance that collaborators listed with a particular email address are actually the person they say they are.

# Acceptance Criteria
A precise description of how/when this issue will be considered closed, e.g.:

- [ ] Registration workflow includes email to address to activate account
- [ ] Email address change happens upon confirming access to new account

**Comment by becky-gilbert** [comment]
While working on other email/SendGrid issues, I noticed that there are quite a few messages being sent to invalid email addresses. In the past month we sent 51 messages to 18 unique email addresses that were undeliverable because the email address was invalid. While it's possible that participants are sometimes intentionally registering their accounts with non-existent emails (e.g. "fake@email.com"), the 18 invalid addresses that we sent messages to over the past month all appear to be caused by typos (e.g. "gmail.con", "gmail.co", "gmai.com", "gmial.com", "hotmail.con", "yahoo.con"). So validating emails would help this situation by alerting users to unintentional typos when entering their email. 

Implementing email validation may also help us reduce the number of spam accounts on the site (though perhaps not, since most spam users are aiming to receive compensation from researchers, which I think they can only receive via email?).
